shopinvader_payment/components/payment_transaction_event_listerner.py

```diff
@@ -15,21 +15,30 @@ def _confirm_and_invalidate_session(self, sale_order):
         if not shopinvader_backend:
             return
         sale_order.action_confirm_cart()
-        response = shopinvader_response.get()
-        response.set_session("cart_id", 0)
-        response.set_store_cache("cart", {})
-        # TODO we should not have to return the last_sale information into the
-        # response, only the id...
-        # This code is an awful hack... We should never have to call
-        # a service implementation from here. That should be the responsibility
-        # of the client to request the cart info to store into ist cache once
-        # once the cache is reset
-        collection = _PseudoCollection("shopinvader.backend", self.env)
-        work = self.work.work_on(collection=collection)
-        setattr(work, "shopinvader_backend", shopinvader_backend)
-        res = work.component(usage="cart")._to_json(sale_order)
-        response.set_store_cache("last_sale", res.get("data", {}))
-        # end of awful code ....
+        try:
+            response = shopinvader_response.get()
+        except RuntimeError:
```

Comment:
Can't we do something more focused that catching RuntimeError?

Reply:
@sbidoul as this code seems in need of being refactored I didn't spend much time onto it :stuck_out_tongue: 
We could make the `get` method fail gracefully and return None if no response/session is there.

Reply:
> We could make the get method fail gracefully and return None if no response/session is there.

That sounds better. Let's not make the code worse than it is.